src/test/ui/associated-type-bounds/duplicate.rs

```diff
@@ -0,0 +1,158 @@
+// compile-fail
+// error-pattern:`()` is not an iterator
```

Comment:
Should probably fix this with `0..1` or something like that where you get these problems. It seems the test doesn't exercise the duplicate error for these cases.

Reply:
Hmm?

Reply:
So if you look at https://github.com/rust-lang/rust/blob/dac096f7d8faf1acddd48fc18216c8c356ab689b/src/test/ui/associated-type-bounds/duplicate.stderr#L228 for example you see that the duplication error doesn't occur. This is because `()` is not an `Iterator`. You may simply want to replace `= ()` or `{}` in various places to something that trivially implements `Iterator`. As an example, `0..1` would fit.